build-support/bin/bootstrap_binary_util.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python2.7
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+# Helper for download_binary.sh to use BinaryUtil to download the appropriate binaries.
+#
+# usage: bootstrap_binary_util.py util_name version filename
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import os
+import re
+import sys
+from functools import reduce
+
+from pants.binaries.binary_util import BinaryRequest, BinaryUtil
+from pants.fs.archive import archiver_for_path
+from pants.option.global_options import GlobalOptionsRegistrar
+from pants.option.options_bootstrapper import OptionsBootstrapper
+from pants.subsystem.subsystem import Subsystem
+
+
+def main():
+  util_name, version, filename = tuple(sys.argv[1:])
```

Comment:
argparse is much more robust and has the benefit of allowing —help. It’s fine for what we have now, but will make it harder to do things like extend this script down the road

Have you worked with it before? If not, I can give the code to get it working! (I’m on phone atm)

Reply:
I have used it once years ago and not long after I ended up fixing the coffeescript argument parsing because I had been thinking about argument parsing so much: https://github.com/jashkenas/coffeescript/pull/3946. I would love the code to do so -- that feels like it's probably half of the point of making this a python script. No rush to do that.

Reply:
Talk to your team-mates, in particular Yi: `Fire.fire(a_method_with_the_obvious_signature)`

Reply:
Argparse is one of my favorite parts of Python. It's so easy to make high quality scripts with it. 

Below is a demo of how it works so that you can see things like optional arguments, which it looks like we don't actually use. Should be easy to convert to this actual program, which I'm happy to help with also.

```python
import argparse

def create_parser():
  parser = argparse.create_parser(description="My script now has fancy CLI options!")
  parser.add_argument("util_name", help="...")
  parser.add_argument("-o", "--optional", help="...")
  return parser

def main():
  args = create_parser().parse_args()
  supportdir = 'bin/{}'.format(args.util_name)
```

If you don't like this line `args = create_parser().parse_args()`, you can have `create_parser()` instead be `parse_args()` and simply return the args. I only set it up this way to make `create_parse()` a pure function & keep side effects in main().